### ISSUE TITLE
The library is incompatible with EEPROM chips 16 MBits and less (AT24c02 - AT24c16) 

### ISSUE THREAD

**Issue by board707** [stefangs/arduino-library-at24cxxx#1]
The library does not take into account that the principle of addressing cells for chips less than 32 Mbits and more is different.
Here is a picture from the datasheet for the AT24c32 chip:
![at24c32addr](https://github.com/stefangs/arduino-library-at24cxxx/assets/50185434/4179bb3b-e6e2-4ab9-a7cc-d2d5e50d8049)
And here is a similar one from the datasheet on AT24c16:
![at24c16addr](https://github.com/stefangs/arduino-library-at24cxxx/assets/50185434/df3b9105-e5ca-4f59-9505-19af58b9e460)
In the first case, the cell address consists of two bytes, and in the second only of one.

Even though in the library you create separate classes for different memory sizes - at24c04,  at24c16, at24c128... the code for writing and reading cells is the same:
```
AT24Cxxx::rawWriteBuffer(uint16_t address, const uint8_t* data, size_t len){
  lastError = 0;
	twoWire->beginTransmission(i2cAddress);
	twoWire->write((uint8_t)((address >> 8) & 0xFF));
	twoWire->write((uint8_t)(address & 0xFF));
```
This code, apparently, uses two bytes for the cell number, that is, it corresponds to the datasheet of EEPROMs of 32 Mbit and more.
For at25c16 and less size chips the library does not work correctly

**Comment by stefangs** [comment]
Thank you, this is correct and a very detailed bug report. I had totally missed this little snag in how the smaller chips are addressed. I assumed they were all addressed in the same manner and as we all know assumptions are the mother of all f*** ups...
I think I have fixed it now, it required bigger changes than I anticipated because of the weird way some of them are addressed. 

I was a bit surprised (but still happy) to get this bug report though, since I don't think I have released this library yet?
